src/panels/config/integrations/ha-integration-card.ts

```diff
@@ -0,0 +1,405 @@
+import {
+  customElement,
+  LitElement,
+  property,
+  html,
+  CSSResult,
+  css,
+  TemplateResult,
+} from "lit-element";
+import { HomeAssistant } from "../../../types";
+import { ConfigEntryExtended } from "./ha-config-integrations";
+import { domainToName } from "../../../data/integration";
+import {
+  ConfigEntry,
+  updateConfigEntry,
+  deleteConfigEntry,
+} from "../../../data/config_entries";
+import { EntityRegistryEntry } from "../../../data/entity_registry";
+import { DeviceRegistryEntry } from "../../../data/device_registry";
+import { showOptionsFlowDialog } from "../../../dialogs/config-flow/show-dialog-options-flow";
+import { showConfigEntrySystemOptionsDialog } from "../../../dialogs/config-entry-system-options/show-dialog-config-entry-system-options";
+import {
+  showPromptDialog,
+  showConfirmationDialog,
+  showAlertDialog,
+} from "../../../dialogs/generic/show-dialog-box";
+import { haStyle } from "../../../resources/styles";
+import "../../../components/ha-icon-next";
+import { fireEvent } from "../../../common/dom/fire_event";
+
+export interface ConfigEntryUpdatedEvent {
+  entry: ConfigEntry;
+}
+
+export interface ConfigEntryRemovedEvent {
+  entryId: string;
+}
+
+declare global {
+  // for fire event
+  interface HASSDomEvents {
+    "entry-updated": ConfigEntryUpdatedEvent;
+    "entry-removed": ConfigEntryRemovedEvent;
+  }
+}
+
+@customElement("ha-integration-card")
+export class HaIntegrationCard extends LitElement {
+  @property() public hass!: HomeAssistant;
+
+  @property() public domain!: string;
+
+  @property() public items!: ConfigEntryExtended[];
+
+  @property() public entityRegistryEntries!: EntityRegistryEntry[];
+
+  @property() public deviceRegistryEntries!: DeviceRegistryEntry[];
+
+  @property() public selectedConfigEntryId?: string;
+
+  protected render(): TemplateResult {
+    if (this.items.length === 1) {
+      return this._renderSingleEntry(this.items[0]);
+    }
+    if (this.selectedConfigEntryId) {
+      const configEntry = this.items.find(
+        (entry) => entry.entry_id === this.selectedConfigEntryId
+      );
+      if (configEntry) {
+        return this._renderSingleEntry(configEntry);
+      }
+    }
+    return this._renderGroupedIntegration();
```

Comment:
Should we make this 2 webcomponents or are you doing this because you have a transition between a list and a specific integration ?

Reply:
Then it would be 3, 1 for keeping the state and 2 for rending grouped/single

Reply:
I think that would be better ?